.github/workflows/site.yml

```diff
@@ -96,7 +107,11 @@ jobs:
           SITE="./checkstyle/target/site"
           LINK="https://${{env.AWS_BUCKET_NAME}}.s3.${{env.AWS_REGION}}.amazonaws.com"
           aws s3 cp $SITE s3://${{env.AWS_BUCKET_NAME}}/$FOLDER/ --recursive
-          echo $LINK/$FOLDER/index.html > message
+          if ["$xdocPath" != "" && "$docPath" != "" ]; then
```

Comment:
please use `-n string` or `-z string` to check that it is/not empty

Reply:
i am not sure 
`          if [ -n "${{$xdocPath}}" && "${{$docPath}}" ]; then
`
is this correct?

Reply:
should work, but it requires some testing

Reply:
i have a question cannt we just update it on checkstyle these changes and test it if something wrong happens revert it?

Reply:
It is not a good option, since it can block this job (and affect other PRs) in case something is wrong. So it is better to test this on fork first (it is tedious, I know).

Reply:
i have made changes as you suggested but for some reasons it's not coming here, so let's try the hard tests as you told on local master then we'll proceed further with this PR.